galley/pkg/source/kube/dynamic/source.go

```diff
@@ -15,103 +15,225 @@
 package dynamic
 
 import (
+	"fmt"
+	"reflect"
+	"sync"
 	"time"
 
 	"istio.io/istio/galley/pkg/runtime"
 	"istio.io/istio/galley/pkg/runtime/resource"
-	"istio.io/istio/galley/pkg/source/kube/client"
 	"istio.io/istio/galley/pkg/source/kube/dynamic/converter"
 	"istio.io/istio/galley/pkg/source/kube/log"
-	"istio.io/istio/galley/pkg/source/kube/schema"
+	sourceSchema "istio.io/istio/galley/pkg/source/kube/schema"
 	"istio.io/istio/galley/pkg/source/kube/stats"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8sRuntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
 )
 
-// source is an implementation of runtime.Source.
-type sourceImpl struct {
-	cfg    *converter.Config
-	ifaces client.Interfaces
-	ch     chan resource.Event
+// source is a simplified client interface for listening/getting Kubernetes resources in an unstructured way.
+type source struct {
+	// Lock for changing the running state of the source
+	stateLock sync.Mutex
 
-	listeners []*listener
-}
+	cfg *converter.Config
 
-var _ runtime.Source = &sourceImpl{}
```

Comment:
Doesn't this still implement the source interface? Can we keep the type assertion?